x-pack/legacy/plugins/siem/server/lib/detection_engine/index/get_index_exists.ts

```diff
@@ -4,15 +4,29 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { IndicesExistsParams } from 'elasticsearch';
-import { CallClusterOptions } from 'src/legacy/core_plugins/elasticsearch';
 import { CallWithRequest } from '../types';
 
 export const getIndexExists = async (
-  callWithRequest: CallWithRequest<IndicesExistsParams, CallClusterOptions, boolean>,
+  callWithRequest: CallWithRequest<
+    { index: string; size: number; terminate_after: number; allow_no_indices: boolean },
+    {},
+    unknown
+  >,
   index: string
 ): Promise<boolean> => {
-  return callWithRequest('indices.exists', {
-    index,
-  });
+  try {
+    callWithRequest('search', {
```

Comment:
You are going to want an `await` here like: `callWithRequest('search', {` otherwise this will not trigger your try/catch block.

Sorry about not having tests for this file. Here is a test file that exercises this:

```ts
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { getIndexExists } from './get_index_exists';

class StatusCode extends Error {
  private statusCode: number;
  constructor(statusCode: number, message: string) {
    super(message);
    this.statusCode = statusCode;
  }
}

describe('get_index_exists', () => {
  test('it should return a false if it encounters a 404', async () => {
    const callWithRequest = jest.fn().mockRejectedValue(new StatusCode(404, 'I am a 404 error'));
    const indexExists = await getIndexExists(callWithRequest, 'some-index');
    expect(indexExists).toEqual(false);
  });

  test('it should reject if it encounters a non 404', async () => {
    const callWithRequest = jest.fn().mockRejectedValue(new StatusCode(500, 'I am a 500 error'));
    await expect(getIndexExists(callWithRequest, 'some-index')).rejects.toThrow('I am a 500 error');
  });
});
```

Reply:
Comparing this logic to:
https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/siem/server/lib/source_status/elasticsearch_adapter.ts#L24

In that logic it uses:

```
if (err.status === 404) {
```

Instead of `statusCode` it uses `status` and also it does this check, `response => response._shards.total > 0,` so does this need to change `statusCode` to `status` or is it wrong? Also do we need to add a final boolean check against the shard total?

Reply:
Hey thank you, you need to use `mockImplementation' to throw an error  or these test above won't work
```
describe('get_index_exists', () => {
  test('it should return a true if no errors', async () => {
    const callWithRequest = jest.fn().mockResolvedValue('');
    const indexExists = await getIndexExists(callWithRequest, 'some-index');
    expect(indexExists).toEqual(true);
  });

  test('it should return a false if it encounters a 404', async () => {
    const callWithRequest = jest.fn().mockImplementation(() => {
      throw new StatusCode(404, 'I am a 404 error');
    });
    const indexExists = await getIndexExists(callWithRequest, 'some-index');
    expect(indexExists).toEqual(false);
  });

  test('it should reject if it encounters a non 404', async () => {
    const callWithRequest = jest.fn().mockImplementation(() => {
      throw new StatusCode(500, 'I am a 500 error');
    });
    await expect(getIndexExists(callWithRequest, 'some-index')).rejects.toThrow('I am a 500 error');
  });
});
```